readme.md

```diff
@@ -62,6 +62,16 @@ const run = async () => {
 })();
 ```
 
+`onFailedAttempt` can return a promise to enable introducing a retry [delay](https://www.npmjs.com/package/delay):
+
+```js
+onFailedAttempt: async (error) => {
+	console.log('waiting for 1s before retrying');
+	await delay(1000)
+},
+```
+
+If `onFailedAttempt` throws, all retries will be aborted.
```

Comment:
This text and example be added to https://github.com/sindresorhus/p-retry/pull/30/files#diff-0730bb7c2e8f9ea2438b52e419dd86c9R96, not here.

You also need to keep https://github.com/sindresorhus/p-retry/blob/master/index.d.ts in sync.

Reply:
> If `onFailedAttempt` throws, all retries will be aborted.

This needs to mention that the original promise will reject with this error.

Reply:
I've updated the docs but I'm not sure what changes to make to the `.d.ts` file, I don't really know any TypeScript, sorry.

Reply:
Just add the docs changes to the `.d.ts` file.

Reply:
I've added the change. Hopefully it's in the right place.